src/core/stream/index.ts

```diff
@@ -393,6 +383,46 @@ export default function Stream({
       warning$
     );
 
+    /**
+     * MediaSource.prototype.endOfStream may throw an exception is one or more
+     * source buffers are being updated. This function allows to handle these exceptions
+     * and direclty retry the end of stream.
+     *
+     * If source buffers are still being updated after 0.5 seconds, then throw.
+     * @param mediaSource
+     * @param countDown
+     */
```

Comment:
Not a fan of this approach :/

Reply:
Still not a fan.

Does it throw when they are being updated even if we're not talking about the last segment? (e.g. after a seek)